packages/react-components/react-tree/src/components/Tree/useTree.ts

```diff
@@ -12,13 +22,168 @@ import type { TreeProps, TreeState } from './Tree.types';
+    requestOpenChange,
+    focusFirstSubtreeItem,
+    focusSubtreeOwnerItem,
     root: getNativeElementProps('div', {
       ref,
+      role: isSubtree ? 'group' : 'tree',
       ...props,
+      ...(isSubtree ? undefined : arrowNavigationProps),
     }),
   };
-};
+}
+
+/**
+ * Create the state required to render the root Tree.
+ *
+ * The returned state can be modified with hooks such as useTreeStyles_unstable,
+ * before being passed to renderTree_unstable.
+ *
+ * @param props - props from this instance of Tree
+ * @param ref - reference to root HTMLElement of Tree
+ */
+function useRootTree(props: TreeProps, ref: React.Ref<HTMLElement>): TreeState {
+  warnIfNoProperPropsRootTree(props);
+  const { openSubtrees: stateOpenSubtrees, defaultOpenSubtrees, onOpenChange } = props;
+  const [openSubtrees, setOpenSubtrees] = useControllableState({
+    state: React.useMemo(() => normalizeOpenSubtreesOrUndefined(stateOpenSubtrees), [stateOpenSubtrees]),
+    defaultState: () => normalizeOpenSubtrees(defaultOpenSubtrees),
+    initialState: [],
+  });
+  const { targetDocument } = useFluent_unstable();
+  const requestOpenChange = useEventCallback((data: TreeOpenChangeData) => {
+    onOpenChange?.(data.event, data);
+    if (!data.event.isDefaultPrevented()) {
+      setOpenSubtrees(updateOpenSubtrees(data, openSubtrees));
+    }
+  });
+  const { treeWalker: treeWalkerRef, root: treeRef } = useTreeWalker(NodeFilter.SHOW_ELEMENT, {
+    acceptNode: filterTreeItemAndSubtree,
+  });
+  const commonState = useSubtree(props, useMergedRefs(ref, treeRef));
+  return {
+    ...commonState,
+    openSubtrees,
+    requestOpenChange,
+    focusFirstSubtreeItem: useEventCallback(target => {
+      const treeWalker = treeWalkerRef.current;
+      if (!treeWalker) {
+        return;
+      }
+      const groupId = target.getAttribute('aria-owns');
+      if (groupId && targetDocument) {
+        const element = targetDocument.getElementById(groupId);
+        if (treeWalker && element) {
+          treeWalker.currentNode = element;
+          const firstTreeItem = treeWalker.firstChild() as TreeItemElement | null;
+          return firstTreeItem?.focus();
+        }
+      }
+    }),
+    focusSubtreeOwnerItem: useEventCallback(target => {
+      const treeWalker = treeWalkerRef.current;
+      if (!treeWalker) {
+        return;
+      }
+      treeWalker.currentNode = target;
+      const group = treeWalker.parentNode() as HTMLElement | null;
+      if (group) {
+        while (treeWalker.previousNode()) {
+          const treeItem = treeWalker.currentNode as TreeItemElement;
+          if (treeItem.getAttribute('aria-owns') === group.id) {
+            return treeItem.focus();
+          }
+        }
+      }
+    }),
+  };
+}
+
+function filterTreeItemAndSubtree(node: Node) {
+  const element = node as HTMLElement & { role: string };
+  return element.role === 'treeitem' || element.role === 'group' ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_REJECT;
```

Comment:
not exactly sure about the name of the enum but I think you want to skip because, if the `treeItem` is wrapped inside another html element you don't want to drill in deeper 

```html
<div role="presentation">
  <div role="treeitem" />
</div>
```

```suggestion
  return element.role === 'treeitem' || element.role === 'group' ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
```